`store.dispatch` does not work in 1.2.2

After migrating to 1.2.2 `store.dispatch` stopped working with the following error message:

```
Argument of type 'ThunkAction<void, CombinedState<{ auth: AuthState; }>, null, Action<string>>' is not assignable to parameter of type 'AnyAction'.
  Property 'type' is missing in type 'ThunkAction<void, CombinedState<{ auth: AuthState; }>, null, Action<string>>' but required in type 'AnyAction'.
```

Dispathing through dispatcher from hook works fine though:

```
  const dispatch = useDispatch();
  ...
  dispatch(logOut());  // OK
```

I can't use React Hooks in the axios interceptor, so I was using this:

```
import store from '../../store';

store.dispatch(logOut());
```

Same problem updating to latest, so here's a code snippet

The `store.ts` config:

```typescript
import { configureStore, Action } from '@reduxjs/toolkit';
import { ThunkAction } from 'redux-thunk';
import { authReducer } from './features/auth/auth.redux';

export const store = configureStore({
    reducer: {
        auth: authReducer,
    },
});

export type AppThunk = ThunkAction<void, RootState, null, Action<string>>;
export type RootState = ReturnType<typeof store.getState>;
```

The thunk in `auth.redux.ts`:

```typescript
export const runLogoutThunk = (deps?: { tokenStore?: ITokenStore }): AppThunk => async dispatch => {
    try {
        const { tokenStore = getTokenStore() } = deps || {};
        const token = tokenStore.getToken();
        tokenStore.removeToken();
        if (token) {
            const authResult = await authApi.logout(token);
        }
    } finally {
        /// HERE IT WILL COMPLAIN
        dispatch(getAuthLogout());
    }
};
```

And the TS error

```typescript
import { store } from '../../store';

export const createDefaultApiService = () => {
    const serverUrl = 'http://localhost:3000';
    const tokenStore = getTokenStore();
    const refreshTokenService = new RefreshTokenService({
        tokenStore: tokenStore,
        refreshTokenUrl: `${serverUrl}/auth/refresh-token`,
    });
    return new ApiService({
        serverUrl,
        refreshTokenService: refreshTokenService,
        tokenStore: tokenStore,
        onAuthFailure: (response): Response => {
           /// TYPESCRIPT IS COMPLAINING HERE ABOUT
           /// ------------------------------------------------------------------
           // error TS2769: No overload matches this call.
           // Overload 1 of 2, '(action: AnyAction): AnyAction', gave the following error.
           // Argument of type 'ThunkAction<void, { auth: AuthState; }, null, Action<string>>' is not assignable to parameter of type 'AnyAction'.
           // Property 'type' is missing in type 'ThunkAction<void, { auth: AuthState; }, null, Action<string>>' but required in type 'AnyAction'.
           // ...
            store.dispatch(runLogoutThunk());
            return response;
        },
    });
};
```

Don't know if it helps getting an idea... I'll try to investigate in the week-end if not found before. Any help appreciated and keep up the fantastic work !

@phryneas, 

> Does it work if you change it like this?
```diff
+ export type AppThunk = ThunkAction<void, RootState, undefined, Action<string>>;
```

Nope

```
error TS2345: Argument of type 'null' is not assignable to parameter of type 'undefined'.
55  await runLogoutThunk({ tokenStore: ts })(dispatch, getState, null);
```                                                                            

Setting it to `unknown` does the job, but I'm not sure about the implications.

--- a/docs/tutorials/advanced-tutorial.md
+++ b/docs/tutorials/advanced-tutorial.md
@@ -541,7 +541,7 @@ Before we go any further, let's add a type declaration we can reuse instead.
 
 export type AppDispatch = typeof store.dispatch
 
-+export type AppThunk = ThunkAction<void, RootState, null, Action<string>>
++export type AppThunk = ThunkAction<void, RootState, unknown, Action<string>>
 ```
 
 The `AppThunk` type declares that the "action" that we're using is specifically a thunk function. The thunk is customized with some additional type parameters:

--- a/etc/redux-toolkit.api.md
+++ b/etc/redux-toolkit.api.md
@@ -81,7 +81,7 @@ export type CaseReducerWithPrepare<State, Action extends PayloadAction> = {
 export type ConfigureEnhancersCallback = (defaultEnhancers: StoreEnhancer[]) => StoreEnhancer[];
 
 // @public
-export function configureStore<S = any, A extends Action = AnyAction, M extends Middlewares<S> = [ThunkMiddleware<S>]>(options: ConfigureStoreOptions<S, A, M>): EnhancedStore<S, A, M>;
+export function configureStore<S = any, A extends Action = AnyAction, M extends Middlewares<S> = [ThunkMiddlewareFor<S>]>(options: ConfigureStoreOptions<S, A, M>): EnhancedStore<S, A, M>;
 
 // @public
 export interface ConfigureStoreOptions<S = any, A extends Action = AnyAction, M extends Middlewares<S> = Middlewares<S>> {

--- a/src/configureStore.ts
+++ b/src/configureStore.ts
@@ -17,10 +17,12 @@ import {
   composeWithDevTools,
   EnhancerOptions as DevToolsOptions
 } from 'redux-devtools-extension'
-import { ThunkMiddleware } from 'redux-thunk'
 
 import isPlainObject from './isPlainObject'
-import { getDefaultMiddleware } from './getDefaultMiddleware'
+import {
+  getDefaultMiddleware,
+  ThunkMiddlewareFor
+} from './getDefaultMiddleware'
 import { DispatchForMiddlewares } from './tsHelpers'
 
 const IS_PRODUCTION = process.env.NODE_ENV === 'production'
@@ -120,7 +122,7 @@ export interface EnhancedStore<
 export function configureStore<
   S = any,
   A extends Action = AnyAction,
-  M extends Middlewares<S> = [ThunkMiddleware<S>]
+  M extends Middlewares<S> = [ThunkMiddlewareFor<S>]
 >(options: ConfigureStoreOptions<S, A, M>): EnhancedStore<S, A, M> {
   const {
     reducer = undefined,

--- a/src/getDefaultMiddleware.ts
+++ b/src/getDefaultMiddleware.ts
@@ -37,7 +37,13 @@ export type ThunkMiddlewareFor<
   ? never
   : O extends { thunk: { extraArgument: infer E } }
   ? ThunkMiddleware<S, AnyAction, E>
-  : ThunkMiddleware<S, AnyAction, unknown>
+  :
+      | ThunkMiddleware<S, AnyAction, unknown>
+      /*
+       * The ThunkMiddleware with a `null` ExtraArgument is here to provide backwards-compatibility.
+       * If you are writing new `ThunkAction` types, going with the `unknown` variant instead would be a better idea.
+       */
+      | ThunkMiddleware<S, AnyAction, null>
 
 /**
  * Returns any array containing the default middleware installed by

--- a/src/getDefaultMiddleware.ts
+++ b/src/getDefaultMiddleware.ts
@@ -28,13 +28,16 @@ interface GetDefaultMiddlewareOptions {
   serializableCheck?: boolean | SerializableStateInvariantMiddlewareOptions
 }
 
-type ThunkMiddlewareFor<S, O extends GetDefaultMiddlewareOptions> = O extends {
+export type ThunkMiddlewareFor<
+  S,
+  O extends GetDefaultMiddlewareOptions = {}
+> = O extends {
   thunk: false
 }
   ? never
   : O extends { thunk: { extraArgument: infer E } }
   ? ThunkMiddleware<S, AnyAction, E>
-  : ThunkMiddleware<S>
+  : ThunkMiddleware<S, AnyAction, unknown>
 
 /**
  * Returns any array containing the default middleware installed by

--- a/src/getDefaultMiddleware.ts
+++ b/src/getDefaultMiddleware.ts
@@ -37,10 +37,9 @@ export type ThunkMiddlewareFor<
   ? never
   : O extends { thunk: { extraArgument: infer E } }
   ? ThunkMiddleware<S, AnyAction, E>
-    /*
-     * The ThunkMiddleware with a `null` ExtraArgument is here to provide backwards-compatibility.
-     */
-  : ThunkMiddleware<S, AnyAction, null> | ThunkMiddleware<S, AnyAction>
+  :
+      | ThunkMiddleware<S, AnyAction, null> //The ThunkMiddleware with a `null` ExtraArgument is here to provide backwards-compatibility.
+      | ThunkMiddleware<S, AnyAction>
 
 /**
  * Returns any array containing the default middleware installed by

--- a/src/getDefaultMiddleware.ts
+++ b/src/getDefaultMiddleware.ts
@@ -37,13 +37,10 @@ export type ThunkMiddlewareFor<
   ? never
   : O extends { thunk: { extraArgument: infer E } }
   ? ThunkMiddleware<S, AnyAction, E>
-  :
-      | ThunkMiddleware<S, AnyAction, unknown>
-      /*
-       * The ThunkMiddleware with a `null` ExtraArgument is here to provide backwards-compatibility.
-       * If you are writing new `ThunkAction` types, going with the `unknown` variant instead would be a better idea.
-       */
-      | ThunkMiddleware<S, AnyAction, null>
+    /*
+     * The ThunkMiddleware with a `null` ExtraArgument is here to provide backwards-compatibility.
+     */
+  : ThunkMiddleware<S, AnyAction, null> | ThunkMiddleware<S, AnyAction>
 
 /**
  * Returns any array containing the default middleware installed by

--- a/type-tests/files/configureStore.typetest.ts
+++ b/type-tests/files/configureStore.typetest.ts
@@ -254,4 +254,20 @@ import thunk, { ThunkMiddleware, ThunkAction, ThunkDispatch } from 'redux-thunk'
     const result2: 'B' = store.dispatch('b')
     const result3: Promise<'A'> = store.dispatch(thunkA())
   }
+  /**
+   * Accepts thunk with `unknown`, `undefined` or `null` ThunkAction extraArgument per default
+   */
+  {
+    const store = configureStore({ reducer: {} })
+    // undefined is the default value for the ThunkMiddleware extraArgument
+    store.dispatch(function() {} as ThunkAction<void, {}, undefined, AnyAction>)
+    // null was previously documented in the redux docs
+    store.dispatch(function() {} as ThunkAction<void, {}, null, AnyAction>)
+    // unknown is the best way to type a ThunkAction if you do not care
+    // about the value of the extraArgument, as it will always work with every
+    // ThunkMiddleware, no matter the actual extraArgument type
+    store.dispatch(function() {} as ThunkAction<void, {}, unknown, AnyAction>)
+    // typings:expect-error
+    store.dispatch(function() {} as ThunkAction<void, {}, boolean, AnyAction>)
+  }
 }

